### DISCUSSION TITLE
Self-hosted runners specific to repository?

### DISCUSSION THREAD

**Discussion by spasam** [community/community#26484] [Actions]
<p>Can self-hosted runners be shared across organization repositories and do we have to create one per repository? It will also be useful to document how to run <strong>config.sh</strong> in non-interactive mode or automatic provisioning. Thanks</p>

**Comment by mscoutermarsh** [comment]
<p>Hi <a class="mention" href="/spasam">@spasam</a>, currently they are per repository. We’ll be working on enabling them for organizations in the future.</p>
<p>Thanks!</p>

**Comment by myoung34** [comment]
<p>For an example on how to do it hands-off (noninteractive):</p>
<p><a href="https://github.com/myoung34/docker-github-actions-runner" rel="nofollow noopener">https://github.com/myoung34/docker-github-actions-runner</a></p>

**Comment by gaieges** [comment]
<p>This is something that we’d very much like as well, likely can’t use github actions + self-hosted runner until then.</p>
<p>Mainly trying to get cached docker builds to run subsequent steps without having to rebuild / re-pull images, which a self-hosted runner would help us do given the way that github actions runs parallel jobs across multiple machines</p>

**Comment by mario-kostadincev** [comment]
<p>Any progress on this? We just moved across fom GitLab and found ourselves in an impossible situation to share self-hosted runners. This has a high impact on our work and if not resolved soon will have to roll-back.</p>

**Comment by Iristyle** [comment]
<p>Agreed that this limitation makes self-hosted runners basically unusable for our needs.</p>
<p>Any ETA on this being changed?</p>

**Comment by jacksonporter** [comment]
<p>It can be done, add multiple runner services on your machine for now.</p>

**Comment by mscoutermarsh** [comment]
<p>It’s actively under development. I don’t have a date yet that I can share. Getting it released as soon as we can.</p>

**Comment by manuyavuz-pointr** [comment]
<p>Hello,</p>
<p>Sorry to bring this up again, but do you now have an ETA given one month has passed since your latest feedback?</p>
<p>It’s not a very big issue to start runner repositories separately, but I don’t think github will be aware of runner idle status across repositories. This means separate repositories can start builds on the same machine, which might cause high resource overload for the same machine.</p>
<p>Is there any workaounds available for the problem I mentioned?</p>
<p>Thanks!</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Organization level runners are now available.</p>
<p><a href="https://github.blog/changelog/2020-04-22-github-actions-organization-level-self-hosted-runners/" rel="nofollow noopener">https://github.blog/changelog/2020-04-22-github-actions-organization-level-self-hosted-runners/</a></p>
